one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -114,6 +121,23 @@ public void onError(String commandName, Exception exception) {
         return payload;
     }
 
+    @RequestMapping(value = "/project-download", produces = "application/zip")
+    public void downloadProject(HttpServletRequest request, HttpServletResponse response) {
```

Comment:
You can test this by using Spring's MockHttpServletRequest (which can be empty) and MockHttpServletResponse. After calling the method, you can get the response bytes from the Mock response. And you should be able to open a ZipInputStream on that and verify a couple entries (no need to verify every single entry - I think verifying 2 of them is enough of a test).